presto-main/src/main/java/com/facebook/presto/server/remotetask/ReactorNettyHttpClient.java

```diff
@@ -0,0 +1,385 @@
+import io.netty.handler.ssl.OpenSsl;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslProvider;
+import reactor.core.Disposable;
+import reactor.core.publisher.Mono;
+import reactor.netty.ByteBufFlux;
+import reactor.netty.channel.MicrometerChannelMetricsRecorder;
+import reactor.netty.http.HttpProtocol;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.http.client.HttpClientResponse;
+import reactor.netty.resources.ConnectionProvider;
+import reactor.netty.resources.LoopResources;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Files;
+import java.security.GeneralSecurityException;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.airlift.security.pem.PemReader.loadPrivateKey;
+import static com.facebook.airlift.security.pem.PemReader.readCertificateChain;
+import static io.micrometer.core.instrument.Clock.SYSTEM;
+import static io.micrometer.jmx.JmxConfig.DEFAULT;
+import static io.netty.handler.ssl.ApplicationProtocolConfig.Protocol.ALPN;
+import static io.netty.handler.ssl.ApplicationProtocolConfig.SelectedListenerFailureBehavior.ACCEPT;
+import static io.netty.handler.ssl.ApplicationProtocolConfig.SelectorFailureBehavior.NO_ADVERTISE;
+import static io.netty.handler.ssl.ApplicationProtocolNames.HTTP_1_1;
+import static io.netty.handler.ssl.ApplicationProtocolNames.HTTP_2;
+import static io.netty.handler.ssl.SslProtocols.TLS_v1_2;
+import static io.netty.handler.ssl.SslProtocols.TLS_v1_3;
+import static io.netty.handler.ssl.SslProvider.JDK;
+import static io.netty.handler.ssl.SslProvider.OPENSSL;
+import static io.netty.handler.ssl.SslProvider.isAlpnSupported;
+import static java.lang.String.format;
+import static java.time.temporal.ChronoUnit.MILLIS;
+
+public class ReactorNettyHttpClient
+        implements com.facebook.airlift.http.client.HttpClient, Closeable
+{
+    private static final Logger log = Logger.get(ReactorNettyHttpClient.class);
+    private static final HeaderName CONTENT_TYPE_HEADER_NAME = HeaderName.of("Content-Type");
+    private static final HeaderName CONTENT_LENGTH_HEADER_NAME = HeaderName.of("Content-Length");
+
+    private final HttpClient httpClient;
+    private final Duration requestTimeout;
+
+    @Inject
+    public ReactorNettyHttpClient(ReactorNettyHttpClientConfig config, InternalCommunicationConfig internalCommunicationConfig)
+    {
+        SslContext sslContext = null;
+        if (internalCommunicationConfig.isHttpsRequired()) {
```

Comment:
Should this be done in the IntenalCommunicationModule: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/InternalCommunicationModule.java#L43

Otherwise I wonder what would happen when `ReactorNettyHttpClient` is bound and the `JettyHttpClient` is not?

Reply:
I am little confused. Are you suggesting.. to initialize the ReactiveNettyHttpClientConfig in INternalCommunicationModue...similar to HttpClient config ?

Reply:
Yeah, I was thinking about that (to keep it in a single place)

Reply:
Done. added changes in internal communications module .